platform/src/main/java/org/stellar/anchor/platform/service/TransactionService.java

```diff
@@ -65,20 +71,25 @@ public GetTransactionResponse getTransaction(String txnId) throws AnchorExceptio
       throw new BadRequestException("transaction id cannot be empty");
     }
 
-    Sep31Transaction txn = txnStore.findByTransactionId(txnId);
-    if (txn == null) {
-      throw new NotFoundException(String.format("transaction (id=%s) is not found", txnId));
+    Sep31Transaction txn31 = txn31Store.findByTransactionId(txnId);
+    if (txn31 != null) {
+      return toGetTransactionResponse(txn31);
     }
 
-    return txn.toPlatformApiGetTransactionResponse();
+    JdbcSep24Transaction txn24 = (JdbcSep24Transaction) txn24Store.findByTransactionId(txnId);
```

Comment:
If Sep24Transaction interface is updated you won't need to cast

Reply:
Yep, we need to cast because we use methods not defined in the Sep24Transaction interface.